src/Transports.AspNetCore/WebSockets/SubscriptionList.cs

```diff
@@ -0,0 +1,228 @@
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// A list of subscriptions represented by <see cref="IDisposable"/> objects.
+/// All members are thread-safe.
+/// <br/><br/>
+/// Upon signalling the cancellation token, all future calls to any method except
+/// <see cref="Dispose"/> will throw an <see cref="OperationCanceledException"/>.
+/// <br/><br/>
+/// It is recommended to call <see cref="Dispose"/> immediately after signalling
+/// the cancellation token.
+/// </summary>
+public sealed class SubscriptionList : IDisposable
+{
+    private readonly CancellationToken _cancellationToken;
+    private readonly Dictionary<string, IDisposable> _subscriptions = new();
+    private readonly object _lock = new();
+
+    /// <summary>
+    /// Initializes a new instance.
+    /// </summary>
+    public SubscriptionList(CancellationToken cancellationToken)
+    {
+        _cancellationToken = cancellationToken;
+    }
+
+    /// <summary>
+    /// Disposes of all active subscriptions.
```

Comment:
```suggestion
    /// Disposes all subscriptions in the list.
```

Formally, they do not have to be active because they can be disposed from the outside.